### ISSUE TITLE
Failing junit tests on windows only

### ISSUE THREAD

**Issue by TanPingZhi** [nus-cs2103-AY2223S1/forum#341]
## Junit test fails most of the time on windows. Ubuntu and MacOS consistently passes
![image](https://user-images.githubusercontent.com/46480755/195294360-f44a29c8-eca9-407d-975b-5cecbd09b7cc.png)

### Here is the error message shown on github
![image](https://user-images.githubusercontent.com/46480755/195287696-b9370d8a-675c-4599-b0c3-8da1ef826fb4.png)

[link to EditCommandTest code](https://github.com/AY2223S1-CS2103T-T17-2/tp/blob/fa8793e23d1057f85e79ef1da9698181a82b44ff/src/test/java/seedu/nutrigoals/logic/commands/EditCommandTest.java#L73)
![image](https://user-images.githubusercontent.com/46480755/195294889-14b9a775-5ac0-4b56-a1bd-5de87d0dab94.png)

[link to FoodComparatorTest code](https://github.com/AY2223S1-CS2103T-T17-2/tp/blob/fa8793e23d1057f85e79ef1da9698181a82b44ff/src/test/java/seedu/nutrigoals/model/meal/FoodComparatorTest.java#L18)
![image](https://user-images.githubusercontent.com/46480755/195294998-3359e935-660d-45e6-925c-a43500063737.png)
[link to src code branch that causes odd test results](https://github.com/AY2223S1-CS2103T-T17-2/tp/tree/test-CI-fail)

**Comment by damithc** [comment]
@TanPingZhi A general note: Tests might not run in the order they are given in the file. Therefore, tests should not depend on each other. If multiple tests change the same data file, the behavior can vary based on the order tests are run.

**Comment by teekaytai** [comment]
Hi! I don't really know what's causing the inconsistent behaviour, but I did notice one thing weird. Regarding this assertion in your `FoodComparatorTest`:

```java
// lunch after breakfast
assertEquals(1, foodComparator.compare(BREAD, APPLE));
```
Seems you intended `BREAD` to be a lunch item and `APPLE` to be a breakfast item. However, when I checked your `TypicalFoods`, they are both tagged `breakfast`. Looks like this was changed in the latest commit to the branch titled [Merge pull request #61 from nicolelim02/update-tag](https://github.com/AY2223S1-CS2103T-T17-2/tp/commit/fa8793e23d1057f85e79ef1da9698181a82b44ff#diff-10489c775072bd082d050446399ed6f49d55cbd3256745fefbfc1e19981dacde).

There were plenty of other changes in that commit too, maybe something there can explain what's happening?

Good luck debugging!
